### PR TITLE
new resources: azurerm_data_lake_analytics_account & azurerm_data_lake_analytics_firewall_rule

### PR DESCRIPTION
**Pull request by katbyte**

### CODE REVIEW

File: azurerm/resource_arm_data_lake_analytics_account.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/services/datalake/analytics/mgmt/2016-11-01/account"
```
Comment (paultyng) [inline azurerm/resource_arm_data_lake_analytics_account.go]: these imports probably need some minor reordering.

File: azurerm/resource_arm_data_lake_analytics_account.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDateLakeAnalyticsAccountCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeAnalyticsAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Date Lake Store creation.")
```
Comment (paultyng) [inline azurerm/resource_arm_data_lake_analytics_account.go]: maybe add the name to this log?

File: azurerm/resource_arm_data_lake_analytics_firewall_rule.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+			},
+		},
+	}
+}
+
+func resourceArmDateLakeAnalyticsFirewallRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeAnalyticsFirewallRulesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Date Lake Analytics Firewall Rule creation.")
```
Comment (paultyng) [inline azurerm/resource_arm_data_lake_analytics_firewall_rule.go]: can you add name to this log?

File: azurerm/resource_arm_data_lake_store.go [hunk 4]
```diff
@@ -2,14 +2,13 @@ package azurerm
...
-	"log"
-	"regexp"
-
 	"github.com/Azure/azure-sdk-for-go/services/datalake/store/mgmt/2016-11-01/account"
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+	"log"
```
Comment (paultyng) [inline azurerm/resource_arm_data_lake_store.go]: weird ordering here

**Comment by paultyng** [comment]
Not sure why Travis failed, hmmm...

File: azurerm/resource_arm_data_lake_analytics_account_test.go [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+%s
+
+resource "azurerm_data_lake_analytics_account" "test" {
+  name                = "acctest%s"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+
+  tier                = "Commitment100AUHours"
+
+  default_store_account_name = "${azurerm_data_lake_store
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_data_lake_analytics_account_test.go]: this test is invalid?

File: azurerm/resource_arm_data_lake_analytics_account_test.go [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+func testAccAzureRMDataLakeAnalyticsAccount_withTags(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_lake_analytics_account" "test" {
+  name                = "acctest%s"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+  
+  default_store_account_name = "${azurerm_data_lake_store
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_data_lake_analytics_account_test.go]: (same here)

File: azurerm/resource_arm_data_lake_analytics_account_test.go [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+func testAccAzureRMDataLakeAnalyticsAccount_withTagsUpdate(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_lake_analytics_account" "test" {
+  name                = "acctest%s"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+  
+  default_store_account_name = "${azurerm_data_lake_store
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_data_lake_analytics_account_test.go]: (same here)

**Comment by katbyte** [comment]
tests pass:

```
[14:14:16] kt@snowbook:~/hashi/..3../terraform-providers/terraform-provider-azurerm$ make fmt; make testacc TEST=./azurerm TESTARGS=-test.run=TestAccAzureRMDataLake
gofmt -w $(find . -name '*.go' |grep -v vendor)
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./azurerm -v -test.run=TestAccAzureRMDataLake -timeout 180m
=== RUN   TestAccAzureRMDataLakeAnalyticsAccount_basic
--- PASS: TestAccAzureRMDataLakeAnalyticsAccount_basic (211.26s)
=== RUN   TestAccAzureRMDataLakeAnalyticsAccount_tier
--- PASS: TestAccAzureRMDataLakeAnalyticsAccount_tier (270.96s)
=== RUN   TestAccAzureRMDataLakeAnalyticsAccount_withTags
--- PASS: TestAccAzureRMDataLakeAnalyticsAccount_withTags (221.77s)
=== RUN   TestAccAzureRMDataLakeAnalyticsFirewallRule_basic
--- PASS: TestAccAzureRMDataLakeAnalyticsFirewallRule_basic (214.68s)
=== RUN   TestAccAzureRMDataLakeAnalyticsFirewallRule_update
--- PASS: TestAccAzureRMDataLakeAnalyticsFirewallRule_update (230.42s)
=== RUN   TestAccAzureRMDataLakeAnalyticsFirewallRule_azureServices
--- PASS: TestAccAzureRMDataLakeAnalyticsFirewallRule_azureServices (214.02s)
=== RUN   TestAccAzureRMDataLakeStoreFirewallRule_basic
--- PASS: TestAccAzureRMDataLakeStoreFirewallRule_basic (140.75s)
=== RUN   TestAccAzureRMDataLakeStoreFirewallRule_update
--- PASS: TestAccAzureRMDataLakeStoreFirewallRule_update (151.38s)
=== RUN   TestAccAzureRMDataLakeStoreFirewallRule_azureServices
--- PASS: TestAccAzureRMDataLakeStoreFirewallRule_azureServices (140.49s)
=== RUN   TestAccAzureRMDataLakeStore_basic
--- PASS: TestAccAzureRMDataLakeStore_basic (132.96s)
=== RUN   TestAccAzureRMDataLakeStore_tier
--- PASS: TestAccAzureRMDataLakeStore_tier (131.89s)
=== RUN   TestAccAzureRMDataLakeStore_withTags
--- PASS: TestAccAzureRMDataLakeStore_withTags (146.22s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	2206.836s
```

**Pull request merged**